Formula/ghc.rb

```diff
@@ -73,22 +82,30 @@ def install
     ENV["LD"] = "ld"
     ENV["PYTHON"] = Formula["python@3.9"].opt_bin/"python3"
 
-    # Build a static gmp rather than in-tree gmp, otherwise all ghc-compiled
-    # executables link to Homebrew's GMP.
-    gmp = libexec/"integer-gmp"
-
-    # GMP *does not* use PIC by default without shared libs so --with-pic
-    # is mandatory or else you'll get "illegal text relocs" errors.
-    resource("gmp").stage do
-      cpu = Hardware::CPU.arm? ? "aarch64" : Hardware.oldest_cpu
-      system "./configure", "--prefix=#{gmp}", "--with-pic", "--disable-shared",
+    args = []
+    on_macos do
+      # Build a static gmp rather than in-tree gmp, otherwise all ghc-compiled
+      # executables link to Homebrew's GMP.
+      gmp = libexec/"integer-gmp"
+
+      # GMP *does not* use PIC by default without shared libs so --with-pic
+      # is mandatory or else you'll get "illegal text relocs" errors.
+      resource("gmp").stage do
+        cpu = Hardware::CPU.arm? ? "aarch64" : Hardware.oldest_cpu
+        system "./configure", "--prefix=#{gmp}", "--with-pic", "--disable-shared",
                             "--build=#{cpu}-apple-darwin#{OS.kernel_version.major}"
-      system "make"
-      system "make", "install"
+        system "make"
+        system "make", "install"
+      end
+
+      args = ["--with-gmp-includes=#{gmp}/include",
+              "--with-gmp-libraries=#{gmp}/lib"]
     end
 
-    args = ["--with-gmp-includes=#{gmp}/include",
-            "--with-gmp-libraries=#{gmp}/lib"]
+    on_linux do
+      args << "--with-gmp-includes=#{Formula["gmp"].opt_include}"
+      args << "--with-gmp-libraries=#{Formula["gmp"].opt_lib}"
+    end
```

Comment:
I thought you couldn't get the `gmp` dependency to work either?

Reply:
I does work but involves more dirty hacks. I'm running the clean version here to remember what the failure was.

Reply:
You could also try to avoid dirty hacks by using the `gmp` that comes bundled with the `ghc` tarball instead, as in my suggestion above.